### ISSUE TITLE
Shape of legend line

### ISSUE THREAD

**Issue by Yannis-ys** [kassambara/survminer#537]
### Expected behavior 
Hi! 
I have found a KM curve like this in a paper. I think the legend of this paper is more clear than the default straight line. I wonder if this type of legend line can also be achieved in survminer. Thank you very much!
![image](https://user-images.githubusercontent.com/52301938/121881103-5fdfc180-cd41-11eb-9cd9-a94ada8c51ca.png)

**Comment by kassambara** [comment]
Yes — this is the small step-shaped legend key from your figure, and you can get it by customizing what `ggsurvplot()` already returns. The survival curves are the first layer of `$plot`, so you swap that layer's legend key for a custom step glyph, overriding only the key and leaving the curve drawing untouched:

```r
library(survminer); library(survival); library(ggplot2); library(grid)

# Draw the legend key of a ggsurvplot's curves as a small step (stair) that
# echoes the Kaplan-Meier shape, instead of the default straight line.
step_legend_key <- function(p) {
  `%||%` <- function(x, y) if (is.null(x)) y else x   # base R has this since 4.4
  draw_key_step <- function(data, params, size) {
    grid::linesGrob(
      x = c(0, 0.45, 0.45, 1), y = c(0.25, 0.25, 0.7, 0.7),
      gp = grid::gpar(col = data$colour %||% "black",
                      lwd = (data$linewidth %||% data$size %||% 0.75) * .pt,
                      lty = data$linetype %||% 1, lineend = "butt"))
  }
  g <- p$plot$layers[[1]]$geom
  p$plot$layers[[1]]$geom <-
    ggplot2::ggproto(paste0(class(g)[1], "Key"), g, draw_key = draw_key_step)
  p
}

fit <- survfit(Surv(time, status) ~ sex, data = lung)
p <- ggsurvplot(fit, data = lung, palette = c("#E7302A", "#2E43F3"),
                legend.title = "Group", legend.labs = c("High", "Low"))
step_legend_key(p)
```

![Step-shaped legend keys that echo the Kaplan-Meier curve](https://i.imgur.com/FMzCuPx.png)

Change the `x`/`y` coordinates inside `draw_key_step()` to tune the step shape (a taller riser, a two-step stair, etc.).

This is now the second recipe in a new "Customization recipes" vignette (dev version), alongside the solid-to-dashed-below-threshold one, so it's easy to find and reuse.

**Comment by kassambara** [comment]
Closing as answered: the `step_legend_key()` recipe above produces the step-shaped legend key from your figure, and it's collected in the new "Customization recipes" vignette for easy reuse.

If a built-in `ggsurvplot()` option for the legend-key shape would be more convenient than the recipe, a thumbs-up or comment here is the signal to add it, and we'll reopen.
